sdk/cosmos/azure-cosmos/azure/cosmos/exceptions.py

```diff
@@ -127,3 +138,13 @@ def _partition_range_is_gone(e):
             and e.sub_status == http_constants.SubStatusCodes.PARTITION_KEY_RANGE_GONE):
         return True
     return False
+
+
+def _container_recreate_exception(e):
+    if (e.status_code in (
+            http_constants.StatusCodes.BAD_REQUEST, http_constants.StatusCodes.NOT_FOUND)) and e.sub_status in (
```

Comment:
Semantically I prefer coupling 1 status code with 1 sub-status code - for example, according to this logic `BAD_REQUEST` and `COLLECTION_RID_MISMATCH` is a valid combination but in actuality isn't.